app/app.go

```diff
@@ -452,6 +452,7 @@ func New(
 		appCodec,
 		keys[datapooltypes.StoreKey],
 		keys[datapooltypes.MemStoreKey],
+		app.GetSubspace(datapooltypes.ModuleName),
```

Comment:
I haven't used Subspace yet, but IIRC, parameters stored in the Subspace cannot be modified by gov proposals. If so, I think we should use Keeper instead of Subspace as genesis parameters of other modules such as `x/staking`: https://github.com/cosmos/cosmos-sdk/blob/v0.42.11/x/staking/keeper/params.go.

Please make sure if it's okay for use to use Subspace.

Reply:
You can also test if the parameter can be updated by gov proposals in your local, by submitting [ParameterChange proposals](https://docs.cosmos.network/v0.44/modules/gov/01_concepts.html#proposal-submission).

Reply:
Ah. I'm just confused. Sorry. I will review other parts again.

Reply:
I'm testing if it is working with submitting ParameterChange proposals. (it seems not working for now and fixing)
As I understand the example you shared (`x/staking`), `k.paramstore` is type of `paramtypes.Subspace` of staking keeper. I will merge this PR after testing parameter change with ParameterChange proposal.